data-contracts/README.md

```diff
@@ -0,0 +1,130 @@
+* [jq](https://jqlang.github.io/jq/)
+* [Gradle](https://gradle.org/install/)
+* JDK 17
+* IDE of choice
+
+> [!Note] When installing and configuring the Confluent CLI, include the Confluent Cloud credentials as environment variables for future use. For instance with bash or zsh, include these export statements:
+> 
+> ```shell
+> export CONFLUENT_CLOUD_API_KEY=<API KEY>
+> export CONFLUENT_CLOUD_API_SECRET<API SECRET>
+> ```
+>
+
+Terraform can use the value of any environment variable whose name begins with `TF_VAR_` as the value of a terraform variable of the same name. For more on this functionality, see the [terraform documentation](https://developer.hashicorp.com/terraform/cli/config/environment-variables#tf_var_name).
+
+Our example requires we set the value the `org_id` variable from Confluent Cloud. This denotes which organization will house the Confluent Cloud assets we are creating. So let's export the Confluent Cloud organization ID to a terraform environment variable.
+
+This command may open a browser window asking you to authenticate to Confluent Cloud. Once that's complete, the result of
+`confluent organization list` is queried by `jq` to extract the `id` of the current organization to which you are authenticated:
+
+```shell
+export TF_VAR_org_id=$(confluent organization list -o json | jq -c -r '.[] | select(.is_current)' | jq '.id')
+```
+
+### Create Assets in Confluent Cloud
+
+From the root of `data-contracts`, run the provided setup script - `setup.sh`. Note, you may need to edit permissions to make this file executable.
+
+```shell
+chmod +x setup.sh
+./setup.sh
+Setup Confluent Cloud and all Gradle Builds for Demo
+-------------------------------------
+Initializing Confluent Cloud Environment...
+......
+......
+......
+......
+BUILD SUCCESSFUL in 2s
+6 actionable tasks: 6 executed
+Watched directory hierarchies: [/Users/sjacobs/code/confluentinc/demo-scene/data-contracts]
+Code Generation Complete
+-------------------------------------
+
+Setup Complete!
+```
+
+Let's have a look at what we've created in Confluent Cloud, we find a new Environment:
+
+![CC Environment](./images/environment.png)
+
+With a Kafka cluster:
+
+![Kafka Cluster](./images/cluster.png)
+
+And Data Contracts:
+
+![Data Contracts](./images/schemas.png)
+
+
+Locally, we also create a `properties` file containing the parameters needed for our Kafka clients to connect to Confluent Cloud. For an example of this 
+`properties` file, see [confluent.properties.orig](shared/src/main/resources/confluent.properties.orig).
+
+> [!NOTE]
+> The file-based approach we're using here is NOT recommended for a production-quality application. Perhaps a secrets manager implementation would be better suited - which the major cloud providers all offer, or perhaps a tool like Hashicorp Vault. Such a tool would also have client libraries in a Maven repository for the JVM applications to access the secrets.
+> 
+
+### Run the Examples
```

Comment:
this section could use instructions on how to run them -- `gradle genrateCode && gradle build` in each directory, and then either the gradle commands to run each version of the app and/or instruction for running from IDE... though I'm confused bcause I see that `setup.sh` should be doing that. for some reason intellij complained about the `Membership` class not being recognized and then when I ran `gradle generateCode` the error went away. maybe an IDE-specific thing, not sure why I ran into this

Reply:
After the setup script, I think it takes IntelliJ a second to reload the project. As much as I like IntelliJ, it often frustrates me.

Reply:
yeah I think that's right. Do you know the incantation to run the app from the command line?

Reply:
Pretty sure you can run the setup script then reload from disk in IntelliJ.